cmake/AwsCFlags.cmake

```diff
@@ -72,7 +72,7 @@ function(aws_set_common_properties target)
         string(REGEX REPLACE "^AWS-C-" "AWS-" EXPORT_DEFINE ${EXPORT_DEFINE})
         string(REPLACE "-" "_" EXPORT_DEFINE ${EXPORT_DEFINE})
 
-        list(APPEND AWS_C_DEFINES -DUSE_IMPORT_EXPORT)
+        list(APPEND AWS_C_DEFINES -DAWS_COMMON_USE_IMPORT_EXPORT)
         list(APPEND AWS_C_DEFINES -D${EXPORT_DEFINE}_EXPORTS)
```

Comment:
This is part of the problem here:

${EXPORT_DEFINE}_EXPORTS needs to be private

AWS_COMMON_USE_IMPORT_EXPORT needs to be public.